packages/fields/src/types/Content/Implementation.js

```diff
@@ -226,9 +227,22 @@ export class Content extends Relationship {
       // in)
       blockTypes: this.blocks.map(block => (Array.isArray(block) ? block[0] : block).type),
 
-      // Key the block options by type to be serialised and passed to the client
       blockOptions: this.blocks
-        .filter(block => Array.isArray(block) && !!block[1])
+        // Normalise the input config so everything is an array
+        .map(block => (Array.isArray(block) ? [block[0], block[1]] : [block, {}]))
+        // Give complex blocks the opportunity to set default values
+        .map(([block, blockConfig]) => {
+          const complexBlockInstance = this.complexBlocks.find(({ type }) => type === block.type);
```

Comment:
Couldn't you do `this.complexBlocks.find((block) => block.constructor === block.implementation)`? and then you wouldn't need to set `this.type`?

Reply:
Yeah, that's one way to do it. I realised I wanted to move `this.type` to the super class anyway as it's important information.

I think both implementations would work.